src/lib/cli_lib/commands.ml

```diff
@@ -22,3 +18,45 @@ let generate_keypair =
       |> Public_key.Compressed.to_base58_check )
       (Rosetta_coding.Coding.of_public_key kp.public_key) ;
     exit 0)
+
+let validate_keypair =
+  Command.async ~summary:"Validate a public, private keypair"
+    (let open Command.Let_syntax in
+    let open Core_kernel in
+    let%map_open privkey_path = Flag.privkey_write_path in
+    Exceptions.handle_nicely
+    @@ fun () ->
+    let validate_transaction keypair =
+      let dummy_payload = Mina_base.Signed_command_payload.dummy in
+      let signature =
+        Mina_base.Signed_command.sign_payload keypair.Keypair.private_key
+          dummy_payload
+      in
+      let message = Mina_base.Signed_command.to_input dummy_payload in
+      let verified =
+        Schnorr.verify signature
+          (Snark_params.Tick.Inner_curve.of_affine keypair.public_key)
+          message
+      in
+      if verified then
+        printf "Verified a transaction using specified keypair\n"
+      else (
+        eprintf "Failed to verify a transaction using the specific keypair\n" ;
+        exit 1 )
+    in
+    let open Deferred.Let_syntax in
+    let%bind () =
+      let password =
+        lazy
+          (Secrets.Keypair.Terminal_stdin.prompt_password "Enter password: ")
+      in
+      match%map Secrets.Keypair.read ~privkey_path ~password with
+      | Ok keypair ->
+          validate_transaction keypair
+      (* TODO: add validation using VRF eval_and_test_public_key *)
```

Comment:
The VRF is effectively a hash function built out of a signature. The only test you can do is to re-evaluate it and check that they're equal, which doesn't seem to give us anything of value. `eval_and_check_public_key` is for snarks, so the additional checks are not relevant for normal use.

Reply:
In other words, you're saying there is no possible universe that a key could properly sign transactions but not evaluate VRFs? I guess if this is the case, then we don't need that extra check.